[5.1]docs(contextmenu): item render event documentation (#1810)
telerik/blazor-docs: components/contextmenu/events.md

# Events

> This article describes the events available in the Telerik Context Menu for Blazor:

Reviewer: ```suggestion
This article describes the events generated by the Telerik Context Menu for Blazor:
```

> This article describes the events generated by the Telerik Context Menu for Blazor:

Reviewer 2: ```suggestion
This article describes the events that are fired by the Telerik Context Menu for Blazor:
```

* [`OnClick`](#onclick)

> * [`OnItemRender`](#onitemrender)

Reviewer 2: The events are sorted alphabetically, which is OK, but then the sections are not.

## OnItemRender

The `OnItemRender` event fires when each Context Menu item renders. It allows you to customize the appearance of an item.

The event handler receives an argument object of type `MenuItemRenderEventArgs` that contains the following properties: 

| Property | Type | Description |
| --- | --- | --- |
| `Item` | `object` | The current item that renders in the Context Menu. |
| `Class` | `string` | The custom CSS class that will be added to the item. |

> >caption Customizing the appearance of the Context Menu items.

Reviewer 2: ```suggestion
>caption Using OnItemRender to customize the appearance of the Context Menu items.
```

````CSHTML
<div class="context-menu-target" style="width:200px; height: 100px; background: yellow; margin-bottom: 50px;">
    Right-click (or tap and hold on a touch device) for a Context Menu.
</div>

> <TelerikContextMenu Data="@MenuItems" Selector=".context-menu-target" OnItemRender="@RenderItemHandler">

Reviewer 2: Consider multi-line declaration

</TelerikContextMenu>

<style>
    .orange {
        background: orange;
    }
</style>

@code {
    public List<ContextMenuItem> MenuItems { get; set; }

    private void RenderItemHandler(MenuItemRenderEventArgs args)
    {
        var item = args.Item as ContextMenuItem;

>         args.Class = "orange";

Reviewer 2: Use a conditional statement to style only 1-2 items, or use several classes and colors.

}
````

## OnItemRender

The `OnItemRender` event fires when each Context Menu item renders. It allows you to customize the appearance of an item.

The event handler receives an argument object of type `MenuItemRenderEventArgs` that contains the following properties: 

| Property | Type | Description |
| --- | --- | --- |
| `Item` | `object` | The current item that renders in the Context Menu. |
| `Class` | `string` | The custom CSS class that will be added to the item. |

>caption Using OnItemRender to customize the appearance of the Context Menu items.

````CSHTML
<div class="context-menu-target" style="width:200px; height: 100px; background: yellow; margin-bottom: 50px;">
    Right-click (or tap and hold on a touch device) for a Context Menu.
</div>

<TelerikContextMenu Data="@MenuItems"
                    Selector=".context-menu-target"
                    OnItemRender="@RenderItemHandler">
</TelerikContextMenu>

<style>
    .orange {
        background: orange;
    }
    .lime {
        background: lime;
    }
</style>

@code {

>     public List<ContextMenuItem> MenuItems { get; set; }

Reviewer 3: ```suggestion
    private List<ContextMenuItem> MenuItems { get; set; }
```
According to code style

    private void RenderItemHandler(MenuItemRenderEventArgs args)
    {
        var item = args.Item as ContextMenuItem;

>         args.Class = item.Items==null ? "orange" : "lime";

Reviewer 2: ```suggestion
        args.Class = item.Items == null ? "orange" : "lime";
```
